block/retriever.go

```diff
@@ -116,6 +117,21 @@ func (m *Manager) ProcessNextDABatch(daMetaData *da.DASubmitMetaData) error {
 		return batchResp.Error
 	}
 
+	if len(batchResp.Batches) == 0 {
+		m.logger.Info("no blocks in batch from DA", "daHeight", daMetaData.Height)
+		return fmt.Errorf("no blocks in batch from DA: %d", daMetaData.Height)
+	}
+
+	lastBatch := batchResp.Batches[len(batchResp.Batches)-1]
+
+	if len(lastBatch.Blocks) == 0 {
+		m.logger.Info("no blocks in last batch from DA", "daHeight", daMetaData.Height)
+		return fmt.Errorf("no blocks in last batch from DA: %d", daMetaData.Height)
+	}
+
+	lastReceivedHeight := lastBatch.Blocks[len(lastBatch.Blocks)-1].Header.Height
+	types.LastReceivedDAHeight.Set(float64(lastReceivedHeight))
```

Comment:
don't think all of this code is necessary.  also you're not guranteed here that the blocks are valid so I don't think that's good. I assume sergi means last valid block but I'm not sure if he means recieved or applied.
if applied, I'd simply say that in the apply block, you simply update the block based on it's source and call it the day. 
@srene what do you think?

Reply:
i think is better to get the da height from here, because if the same block is received from p2p its not going to be added to the cache and not applied from da, so this info will be lost. if the block is not valid i dont think is relevant in terms of debugging the syncing status.

Reply:
do we care if they are valid or not? I think we care if we received them or not